jakt.nix

```diff
@@ -1,13 +1,26 @@
-{ pkgs ? import <nixpkgs> { } }:
-pkgs.mkShell
-{
-  name = "jakt";
+{ stdenv
+, lib
+, gitignoreSource
+, buildInputs
+, nativeBuildInputs
+, clang
+, fetchFromGitHub
+}:
 
-  nativeBuildInputs = with pkgs; [
-    pkgconfig
-    cmake
-    ninja
-    python3
-    clang_15
+stdenv.mkDerivation (finalAttrs: let
+  serenity = fetchFromGitHub {
+    owner = "serenityos";
+    repo = "serenity";
+    rev = "05e78dabdbceea46bae7dca52b63dc0a115e7b52"; # latest at the time
```

Comment:
Do we really need to pin the Serenity version? generally the only expectation is that jakt main builds with serenity master. This means that the env will break without notice at any time, and anyone using this will have to update this rev and hash any time they want to spin up an env again.

Reply:
To ensure reproducible builds, yes we do need to pin the Serenity version. I understand that this is a new change, however consider the possibility that Serenity could have a breaking change in master that would cause jakt to fail to build without any warning. By removing a pinned serenity version, we would have to remove support all support for nix builds.

Reply:
I realize that hardcoding a git revision here isn't exactly the prettiest solution to this problem. Perhaps git submodules could be used?

Reply:
I still don't like pinning this, but that's the nature of nix I suppose. If something in serenity breaks jakt, then we should fix it rather than ignoring it until the next time someone bothers to update this hash. Git submodules are significantly worse than any other solution, especially for working across repos.